libbpf-tools/Makefile

```diff
@@ -111,6 +125,16 @@ $(OUTPUT)/%.bpf.o: %.bpf.c $(LIBBPF_OBJ) $(wildcard %.h) $(ARCH)/vmlinux.h | $(O
 		     -I$(ARCH)/ $(INCLUDES) -c $(filter %.c,$^) -o $@ &&      \
 	$(LLVM_STRIP) -g $@
 
+ifdef ENABLE_MIN_CORE_BTFS
+$(BTF_HUB_ARCHIVE):
+	$(call msg,GIT CLONE,$@)
+	$(Q)git clone -q https://github.com/aquasecurity/btfhub-archive/ $(BTF_HUB_ARCHIVE)
+
+$(OUTPUT)/min_core_btf_tar.o: $(patsubst %,$(OUTPUT)/%.bpf.o,$(APPS)) $(BTF_HUB_ARCHIVE) | bpftool
+	$(Q)cd $(BTF_HUB_ARCHIVE) && git pull
+	$(Q)$(MAKE) -f Makefile.btfgen
+endif
+
```

Comment:
this has a downside of not checking if anything changed about btf-hub. I did this locally and it seems to work well plus it defines "btf-hub" command to fetch it, even if we don't specify ENABLE_MIN_CORE_BTFS.

```diff
diff --git a/libbpf-tools/Makefile b/libbpf-tools/Makefile
index caf39b6409c1..6426262ea395 100644
--- a/libbpf-tools/Makefile
+++ b/libbpf-tools/Makefile
@@ -125,13 +125,13 @@ $(OUTPUT)/%.bpf.o: %.bpf.c $(LIBBPF_OBJ) $(wildcard %.h) $(ARCH)/vmlinux.h | $(O
                     -I$(ARCH)/ $(INCLUDES) -c $(filter %.c,$^) -o $@ &&      \
        $(LLVM_STRIP) -g $@

-ifdef ENABLE_MIN_CORE_BTFS
-$(BTF_HUB_ARCHIVE):
-       $(call msg,GIT CLONE,$@)
-       $(Q)git clone -q https://github.com/aquasecurity/btfhub-archive/ $(BTF_HUB_ARCHIVE)
-
-$(OUTPUT)/min_core_btf_tar.o: $(patsubst %,$(OUTPUT)/%.bpf.o,$(APPS)) $(BTF_HUB_ARCHIVE) | bpftool
+btf-hub: force
+       $(call msg,GIT,$@)
+       $(Q)[ -d "$(BTF_HUB_ARCHIVE)" ] || git clone -q https://github.com/aquasecurity/btfhub-archive/ $(BTF_HUB_ARCHIVE)
        $(Q)cd $(BTF_HUB_ARCHIVE) && git pull
+
+ifdef ENABLE_MIN_CORE_BTFS
+$(OUTPUT)/min_core_btf_tar.o: $(patsubst %,$(OUTPUT)/%.bpf.o,$(APPS)) btf-hub | bpftool
        $(Q)$(MAKE) -f Makefile.btfgen
 endif

@@ -157,6 +157,9 @@ install: $(APPS) $(APP_ALIASES)
        $(Q)$(INSTALL) $(APPS) $(DESTDIR)$(prefix)/bin
        $(Q)cp -a $(APP_ALIASES) $(DESTDIR)$(prefix)/bin

+.PHONY: force
+force:
+
```

This will always try to fetch latest btfhub version before building.